File: java-order/src/main/java/com/veben/javatokotlin/java/order/ext/db/JpaOrderRepository.java
package com.veben.javatokotlin.java.order.ext.db;

import com.veben.javatokotlin.java.order.domain.PassedOrder;
import com.veben.javatokotlin.java.order.domain.OrderRepository;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Repository
public interface JpaOrderRepository extends OrderRepository, JpaRepository<PassedOrder, UUID> {

    default Set<PassedOrder> findAllOrders() {
        return new HashSet<>(findAll());
    }
}

File: kotlin-order/src/main/kotlin/com/veben/javatokotlin/kotlin/order/ext/db/JpaOrderRepository.kt
package com.veben.javatokotlin.kotlin.order.ext.db

import com.veben.javatokotlin.kotlin.order.domain.PassedOrder
import com.veben.javatokotlin.kotlin.order.domain.OrderRepository
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*
import kotlin.collections.HashSet
import kotlin.collections.Set

@Repository
interface JpaOrderRepository : OrderRepository, JpaRepository<PassedOrder, UUID> {

    @JvmDefault
    override fun findAllOrders(): Set<PassedOrder> {
        return HashSet(findAll())
    }
}